infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/LogReplicationClientRouter.java

```diff
@@ -181,6 +181,22 @@ public IClientRouter addClient(IClient client) {
     public <T> CompletableFuture<T> sendRequestAndGetCompletable(
             @Nonnull RequestPayloadMsg payload,
             @Nonnull String nodeId) {
+        return sendRequestAndGetCompletable(payload, nodeId, timeoutResponse);
+    }
+
+    /**
+     * Send a request message and get a completable future to be fulfilled by the reply within
+     * supplied timeout response time.
+     *
+     * @param payload
+     * @param <T> The type of completable to return.
+     * @return A completable future which will be fulfilled by the reply,
+     * or a timeout in the case there is no response within the allotted time.
+     */
+    public <T> CompletableFuture<T> sendRequestAndGetCompletable(
```

Comment:
This will send the message immediately but wait for a longer time for the response.  Instead, hadnt we decided to wait for this time before re-sending?

Reply:
I think we had found that either was suitable, however doing it this way we avoid introducing a sleep in our sender while getting the same effect of the added back pressure

Reply:
hmm... I may be mistaken but I dont remember deciding this.  Backoff essentially means waiting before retrying...